### PR TITLE
Increase unit test coverage

### PR DESCRIPTION
**Pull request by animesh371**
## Issue

https://github.com/harmony-one/harmony/issues/316

#### Test Coverage Data

* Before
total:									(statements)				49.1%
* After
total:									(statements)				54.3%

### CODE REVIEW

**Comment by LeoHChen** [comment]
Can one of the admins verify this patch?

File: drand/drand_leader_msg_test.go [hunk 1]
```diff
@@ -24,3 +24,21 @@ func TestConstructInitMessage(test *testing.T) {
...
+	priKey, _, _ := utils.GenKeyP2P("127.0.0.1", "9902")
+	host, err := p2pimpl.NewHost(&leader, priKey)
+	if err != nil {
+		test.Fatalf("newhost failure: %v", err)
+	}
+	dRand := New(host, "0", []p2p.Peer{leader, validator}, leader, nil, true)
+	dRand.blockHash = [32]byte{}
+	msg := dRand.constructCommitMessage([32]byte{}, []byte{})
+
+	if len(msg) != 191 {
```
Comment (LeoHChen) [inline drand/drand_leader_msg_test.go]: to compare the length sometimes fail in travis_checker?  do you know of anything else makes more sense to check?

File: drand/drand_test.go [hunk 2]
```diff
@@ -22,3 +27,159 @@ func TestNew(test *testing.T) {
...
+	dRand := New(host, "0", []p2p.Peer{leader, validator}, leader, nil, true)
+
+	if !dRand.IsLeader {
+		test.Error("dRand should belong to a leader")
+	}
+
+	newPeer := p2p.Peer{IP: "127.0.0.1", Port: "9907"}
+	countValidatorPeers := dRand.AddPeers([]*p2p.Peer{&newPeer})
+
+	if countValidatorPeers != 1 {
```
Comment (LeoHChen) [inline drand/drand_test.go]: can you also verify the total number of peers after the AddPeers call?

File: drand/drand_test.go [hunk 3]
```diff
@@ -22,3 +27,159 @@ func TestNew(test *testing.T) {
...
+	host, err := p2pimpl.NewHost(&leader, priKey)
+	if err != nil {
+		test.Fatalf("newhost failure: %v", err)
+	}
+	dRand := New(host, "0", []p2p.Peer{leader, validator}, leader, nil, true)
+
+	_, pubKey1 := utils.GenKey("127.0.0.1", "5555")
+	_, pubKey2 := utils.GenKey("127.0.0.1", "6666")
+
+	if dRand.UpdatePublicKeys([]*bls.PublicKey{pubKey1, pubKey2}) != 2 {
```
Comment (LeoHChen) [inline drand/drand_test.go]: any better way to make make the public keys are updated properly?

File: drand/drand_validator_msg_test.go [hunk 4]
```diff
@@ -24,3 +24,22 @@ func TestConstructCommitMessage(test *testing.T) {
...
+	priKey, _, _ := utils.GenKeyP2P("127.0.0.1", "9902")
+	host, err := p2pimpl.NewHost(&leader, priKey)
+	if err != nil {
+		test.Fatalf("newhost failure: %v", err)
+	}
+	dRand := New(host, "0", []p2p.Peer{leader, validator}, leader, nil, true)
+	dRand.blockHash = [32]byte{}
+	msg := dRand.constructInitMessage()
+
+	if len(msg) != 93 {
```
Comment (LeoHChen) [inline drand/drand_validator_msg_test.go]: to check the length is not always reliable, any better way?

File: drand/drand_test.go (hunk 2 above)
Comment (animesh371) [reply drand/drand_test.go]: Done

File: drand/drand_validator_msg_test.go (hunk 4 above)
Comment (animesh371) [reply drand/drand_validator_msg_test.go]: I think a better way would be to unMarshall & check for the payload. Let me know your thoughts.

File: drand/drand_leader_msg_test.go (hunk 1 above)
Comment (animesh371) [reply drand/drand_leader_msg_test.go]: We can unmarshall the messagepayload and check for the msg type. Let me know your thoughts.

File: drand/drand_validator_msg_test.go (hunk 4 above)
Comment (LeoHChen) [reply drand/drand_validator_msg_test.go]: make sense to me.

File: drand/drand_leader_msg_test.go (hunk 1 above)
Comment (LeoHChen) [reply drand/drand_leader_msg_test.go]: sounds good.

File: drand/drand_test.go (hunk 3 above)
Comment (animesh371) [reply drand/drand_test.go]: Along with the count, I am comparing the hexString of publicKeys which are updated.

**Comment by animesh371** [comment]
@LeoHChen I have addressed the review comments. Have a look

**Comment by LeoHChen** [comment]
I approved your PR.  Please rebase it on top of the tree and provide the updated test coverage data.  I'll merge it after you update this PR.

**Comment by LeoHChen** [comment]
@animesh371 , the travis checker failed with the following errors.
    drand_validator_msg_test.go:27: Commit message is not constructed in the correct size: 190

Can you fix that test as well?  Thanks.

**Comment by animesh371** [comment]
Hi @LeoHChen, I have fixed the issue.

**Pull request merged**